codec/src/test/java/io/netty/handler/codec/ByteToMessageDecoderTest.java

```diff
@@ -345,4 +344,177 @@ public CompositeByteBuf addComponent(boolean increaseWriterIndex, ByteBuf buffer
             assertEquals(0, in.refCnt());
         }
     }
+
+    private static abstract class PartialCumulationDecoder extends ByteToMessageDecoder
+    {
+        final int messageSize;
+        int cumulations;
+
+        private PartialCumulationDecoder(int messageSize) {
+            this.messageSize = messageSize;
+            setCumulator(new PartialCumulator());
+        }
+
+        private class PartialCumulator implements Cumulator {
+            @Override
+            public ByteBuf cumulate(ByteBufAllocator alloc, ByteBuf cumulation, ByteBuf in) {
+                ++cumulations;
+                if (!cumulation.isReadable()) {
+                    return in;
```

Comment:
It's actually unnecessary for our usage to call release in either implementation, but it may be that external users of the API pass pooled empty buffers as the cumulation parameter, so we want to be safe to that.

I'm absolutely happy to introduce a release call here for symmetry, though.

Reply:
I see.... I just wanted to understand if there is any "change" in behaviour related to who is responsible to release the buffer. From what I understand there is no change here really, correct ?

Reply:
Correct.  At least, not with normal usage, and in this case.  However, to be absolutely certain I've not been unclear elsewhere, IF the cumulator chooses not to fully copy the bytes (if it is message aware), then the buffer remains the responsibility of the caller (which is necessary, since it must consume the bytes).  This is only ever the case if you choose to use such a cumulator, though.

Reply:
got it.... Please add the release call here then as well. Just to make it consistent